visidata/sheets.py

```diff
@@ -1208,6 +1208,21 @@ def _async_deepcopy(newlist, oldlist):
     _async_deepcopy(ret, rowlist)
     return ret
 
+@Sheet.api
+def reload_or_replace(sheet):
+    sheet.preloadHook()
+    if isinstance(sheet.source, visidata.Path) and \
+       sheet.source.is_url() and sheet.source.scheme != 'file':  #2825
+        #retrieve data again, because the earlier data saved in sheet.source may be outdated
+        vs = vd.openSource(Path(sheet.source.given))
+        if type(vs) != type(sheet):  #new data may have a different filetype
```

Comment:
I think there's a more Pythonic way to do this.  `is` or `isinstance`?  But I'm not sure, and I don't think it's a big deal.